conf/modules.pna.config

```diff
@@ -0,0 +1,231 @@
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Config file for defining DSL2 per module options and publishing paths
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Available keys to override module options:
+        ext.args   = Additional arguments appended to command in module.
+        ext.args2  = Second set of arguments appended to command in module (multi-tool modules).
+        ext.args3  = Third set of arguments appended to command in module (multi-tool modules).
+        ext.prefix = File name prefix for output files.
+----------------------------------------------------------------------------------------
+*/
+
+
+process {
+
+    withName: PIXELATOR_PNA_AMPLICON {
+        ext.args = { [
+                params.pna_amplicon_mismatches ? "--mismatches ${params.pna_amplicon_mismatches}": '',
+                params.pna_amplicon_remove_polyg ? "--remove-polyg": '',
+                params.pna_amplicon_quality_cutoff ? "--quality-cutoff ${params.pna_amplicon_quality_cutoff}": '',
+            ].join(' ').trim()
+        }
+
+        publishDir = [
+            [
+                path: { "${params.outdir}/pixelator" },
+                mode: params.publish_dir_mode,
+                pattern: 'amplicon/*.amplicon.fq.zst',
+                saveAs: { (params.save_pna_amplicon_reads || params.save_pna_all) ? it : null }
+            ],
+            [
+                path: { "${params.outdir}/pixelator/logs/${meta.id}" },
+                mode: params.publish_dir_mode,
+                pattern: "*.log"
+            ],
+            [
+                path: { "${params.outdir}/pixelator" },
+                mode: params.publish_dir_mode,
+                pattern: '**/*.{report,meta}.json',
+                saveAs: { params.save_json || params.save_pna_all ? it : null }
+            ]
+        ]
+    }
+
+    withName: PIXELATOR_PNA_DEMUX {
+        ext.args = {
+            [
+                (params.pna_demux_mismatches instanceof Integer) ? "--mismatches ${params.pna_demux_mismatches}": '',
```

Comment:
Similarly, I would deal with this params here only with existence and check the type with the schema

Reply:
I encountered a similar issue recently, so I fully agree about this test.